correct some typos, grammer, formatting, and omissions

bill-auger opened the pull request.

bill-auger committed:
fix typos and formatting in doc/install.rst

bill-auger committed:
fix typos and add dependencies to doc/development.rst

bill-auger committed:
fix typos, grammar, and add missing keys in doc/configuration.rst

  * added documentation for key UPLOAD_FOLDER_PATH
  * added documentation for key GITOLITE_CONFIG

pypingou committed:
Fix the links in the templates to account for the namespaces

pypingou committed:
Adjust unit-tests for the namespace

pypingou committed:
Adjust the API routes for the namespaces and the ForkRepoForm used in the API

pypingou committed:
More fixes to make create project through the API working with namespace

pypingou committed:
Drop the RepoConverter that we use to simulate namespaces at first

At first, we supported what we called 'fake' namespace that were just
allowing projects to have a '/' in their name.
Later it was figured out that this approach wasn't valid and thus, we
moved onto creating 'real' namespaces.
This commit removes the mechanism allowing us to use 'fake' namespaces.

pypingou committed:
Fix docstring

pypingou committed:
Reword the title for the namespace in the ProjectForm

pypingou committed:
Account for namespaces in the path to the git repo

pypingou committed:
Include the namespace in project.name (if there is a namespace)

pypingou committed:
Include the repo's namespace in the header of the repo pages

pypingou committed:
Include the project's namespace in the redirect URLs when working on issues

pypingou committed:
Allow setting a namespace when creating a project

pypingou committed:
Ensure there is an user before looking at its groups

pypingou committed:
For some reasons namespaces can sometime be 'None' where we expect None

pypingou committed:
Add an alembic revision to add the namespace field to the project table

pypingou committed:
Publish the namespace in the JSON representation of the project

pypingou committed:
Include the namespace in our set_variables method called before_request

pypingou committed:
Move the logic for the backward compatibility URLs to set_variables

pypingou committed:
Drop spaces in the title/template

pypingou committed:
Adjust unit-test now that the namespace is included in the JSON of a project

pypingou committed:
Rework get_repo_name to use rsplit rather than plain split

pypingou committed:
Rework the get_username method to not rely on config['FORK_FOLDER']

pypingou committed:
Add a get_repo_namespace to extract the namespace from the path of the git repo

pypingou committed:
Add namespace supports for all of our git hooks

pypingou committed:
Fix style and specify the namespace in the error message

pypingou committed:
Specify the namespaces in the unit-tests

pypingou committed:
Rework the get_repo_namespace method in pagure.lib.git

pypingou committed:
Adjust pagure-ci and pagure-webhook to account for namespaces

pypingou committed:
Let's get the namespace by rsplit rather than split

pypingou committed:
Expand unit-tests for pagure.lib.git.get_username

pypingou committed:
Add unit-tests for pagure.lib.git.get_namespace

pypingou committed:
Adjust the regex validating project names

pypingou committed:
Fix typo in the edit_tag.html template

pypingou commented on doc/configuration.rst:
```diff
@@ -120,17 +117,24 @@ documentation of the projects are stored.
 
 
+FORK_FOLDER
+~~~~~~~~~~~
```
This is no longer used

pypingou committed:
Add the opportunity to list all issues, opened and closed

Fixes https://pagure.io/pagure/issue/1288

pypingou committed:
Add the opportunity to list all PRs, opened and closed

Fixes https://pagure.io/pagure/issue/1289

pypingou committed:
Order the issues, most recent first, all the time

Fixes https://pagure.io/pagure/issue/1112

pypingou committed:
Add the possibility to segment the issues and PRs retrieved from the DB

This is required to paginating the results

pypingou committed:
Retrieve and set offset, page and limit from the URLs if any are provided

pypingou committed:
Rework the view_issues endpoint to support pagination

pypingou committed:
Rework the request_pulls endpoint to support pagination

pypingou committed:
Adjust the pagination_link macro to be more flexible

This way the url generated for next/previous pages will keep the
arguments they have and will just adjust the page one

pypingou committed:
Add pagination supports to the issues and requests templates

Fixes https://pagure.io/pagure/issue/1290

pypingou committed:
Fix displaying the status of the issues shown

pypingou committed:
Adjust unit-tests for the change in ordering of the issues returned/shown

pypingou committed:
Add unit-tests for showing all the tickets and checking the title given

pypingou committed:
Adjust the spacing around the status of the PRs shown

pypingou committed:
Fix filtering the issues by status

pypingou committed:
Adjust unit-tests for the change in HTML in the requests template

pypingou committed:
Increase the number of blacklisted project names

pypingou committed:
Add a form used when adding a new report

pypingou committed:
Add a reports property to the projects

That's the JSON blob that will contain the information about the
filtering to do for each report (as well as the name of the report).

pypingou committed:
Add a method to save the filters applied in a given URL to the database

This is the method saving the report with a given name

pypingou committed:
Add the UI endpoint to save a report to the database

pypingou committed:
Add the button to save the current view as a report

pypingou committed:
Add the endpoint to view a given report

pypingou committed:
Add the button in the issue page to load a certain report

pypingou committed:
Fix indentation in the repo_master template

pypingou committed:
Drop the two console.log() call in edit_file.html

pypingou committed:
Add missing namespace arguments and fix typos pointed out by @vivekanand1101

pypingou committed:
Adjust the is_watching method for namespaces, thanks @vivekanand1101

pypingou committed:
Add missing namespaces arguments in the template/urls, thanks @vivekanand1101

pypingou committed:
Fix typo, dots aren't comas

pypingou committed:
Adjust the order of the arguments in the links and some functions

This way we are consistently using: reponame, username, namespace

pypingou committed:
Specify the allowed namespaces when creating a project via the API

pypingou committed:
Fix the ProjectForm to be allowed to return None, not just unicode objects

pypingou committed:
Fix the pagination links when the url has no arguments at the beginning

pypingou committed:
Store in one place the number of items per page (tickets/PRs)

pypingou committed:
Blacklist `user` project

pypingou committed:
Add the alembic migration script to add the report field to the project table

pypingou committed:
Add support to our local markdown processor for ~~striked~~

Fixes https://pagure.io/pagure/issue/1261

vivekanand1101 committed:
Fix typo in the search_projects while filtering for namespace in db query

ryanlerch committed:
replaced the 0.12.3 versions of selectise to use the standalone version

ryanlerch committed:
fixed emojicomtepte issue

ryanlerch committed:
added missing emojicomplete file

ryanlerch committed:
fix openiconic not loading

pypingou committed:
Fix atwho

This fixes @<username> and #<id> by bringing back the right version of
jquery.carret as well as updating jquery.atwho to 1.5.1

pypingou committed:
Release 2.5

We invert the order of two alembic migrations as otherwise the migration
for the namespaces will fail.

pypingou committed:
Fix displaying the branches available for a new PR in the requests template

We have changed the format of the data returned by the endpoint listing
the branches ready to be in a PR so that it also includes branches that
are already in a PR (the info is then displayed on the front page of the
project).
However, this same internal API endpoint is also called in the page
listing all the pull-requests and this page had not been adjusted for
this data format change.

This commit fixes that.

Fixes https://pagure.io/pagure/issue/1310

pypingou committed:
Add a milestone column to the issues table

pypingou committed:
Add alembic migration script adding a milestone column to the issues table

pypingou committed:
Add support to edit the issue's milestone on pagure.lib.edit_issue

pypingou committed:
Implement setting and editing milestone on issues

pypingou committed:
Add the possibility to search issues by milestones

pypingou committed:
Fix the roadmap endpoint to properly group the issues

pypingou committed:
Fix the roadmap/list buttons

pypingou committed:
Drop the unplanned since it's no longer a feature

pypingou committed:
Adjust the unit-tests for the new behavior of issues' milestones

pypingou committed:
Include the milestone in the JSON representation of an issue

pypingou committed:
Adjust unit-tests for the change in JSON representation of an issue

pypingou committed:
Adjust the documentation about using the roadmap feature

pypingou committed:
Some more adjustments to the unit-tests for the change when editing an issue

vivekanand1101 committed:
use username for pygit2 signature in case fullnameisn't there while creating new projects

pypingou committed:
Rework a little more pagure-ci

Use the request UID to retrieve its object in the database and from
there forward to jenkins the right repo, the one containing the new
changes, not the one where we want these changes to go.

jasontibbitts committed:
Fix grammar on issues page

The current issues page will show something like:
  100 Open Issues (on 140)

"on" should be "of".  "out of" also works, but I've gone with the
former.

jasontibbitts committed:
Also fix grammar on pull request page.

pypingou committed:
Add a project's setting allowing to set the default privacy status

Fixes https://pagure.io/pagure/issue/1285

pypingou committed:
Fix the alembic script to drop the milestone column upon downgrade

bookwar committed:
[docs] Fix typo in using_docs example

We need to clone initial pagure repo to build docs from sources, not the
docs one.

pypingou committed:
Allow editing the privacy setting in the issue's metadata

Fixes https://pagure.io/pagure/issue/1286

pypingou committed:
Fix unit-tests due to the changes made in #1317

pypingou committed:
Fix typos on the roadmap doc pointed out by @vivekanand1101

pypingou committed:
Add a _notifications column to the projects table

pypingou committed:
Add the possibility to send notification for issues/PRs

With this commit, admins can specify arbitrary email addresses to notify
when a ticket/issue or a pull-request is created, updated or closed.

This can be used, for example to notify a mailing-list of ticket/issue
update.

Fixes https://pagure.io/pagure/issue/1258

pypingou committed:
Fix viewing plugins and returning to the proper page on projects with a namespace

pypingou committed:
Ensure the login._check_session_cookie is always the first @before_request

This in order to be sure that flask.g.fas_user is set to something when
it should and thus that the order @before_request functions can rely on
it.

pypingou committed:
Add a custom validator allowing to enter multiple emails in one field

pypingou committed:
Inform the user when the data entered wasn't valid

pypingou committed:
Misc small fixes fixing running the unit-tests

pypingou committed:
Allow specifying the root folder to pagure.lib.git.get_repo_namespace

This allows specifying a different root folder for projects, tickets
and requests.

Fixes https://pagure.io/pagure/issue/1321

jeremycline committed:
Update the issues tag button to match Pull Requests

The issues tags button for closed issues is called "Close", while the
pull requests page uses "Closed". This updates the issues page template
to match the pull requests page.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

pypingou committed:
Fix invalid indentation pointed out by @vivekanand1101

pypingou committed:
Small pep8 fix pointed out by @vivekanand1101

pypingou committed:
Release 2.6

lubomir committed:
Use long dash in footer instead of two short ones

This way there is no gap in between them.

lubomir committed:
Update milestone description in Settings

The milestones are no longer based on tags, so the description should be
updated to correctly describe how they work.

vivekanand1101 committed:
remove unused import from lib/__init__

jeremycline committed:
Create a Vagrantfile and Ansible role for Pagure development

This is by no means a complete environment, but it's a good place to
start.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

farhaanbukhsh committed:
Fix NoneType error when pagure-ci form is inactively updated first time

When pagure-ci form is updated for the first time and the  active
 box is not checked it threw error, because the ci_hook was not
generated that time.

jeremycline committed:
Fix minor typos in configuration documentation

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

jeremycline committed:
Use context managers to ensure files are closed

There were several places where context managers were not being used and
files were potentially not being closed.

I started investigating this since the unit tests are not passing if the
code directory is mounted with SSHFS. The test setUp failed to remove
directories with ``shutil.rmtree`` with "directory not empty". Looking
at the directories, there were .fuse_hidden* files so I assumed the
file handles were being leaked. The tests are still failing so I've
either missed something in my search or that was not the problem, but I
thought this would be a good change in any case.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

jeremycline committed:
Remove hardcoded hostnames in unit tests

This commit removes the hardcoded 'https://pagure.org/' in unit test
assertions about error messages.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

jeremycline committed:
Update the setup.py

The setup.py included links that either returned HTTP 404 or did not
have any content. Additionally, it did not include any classifiers, so I
added a few that seemed reasonable to me.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

vivekanand1101 committed:
Adjust update_tickets_from_git to add milestones for issues as well

Commit:
Searches given word in title of issues.

Changes according to pep8
Docstring added for search pattern and pep8 changes
search bar and button added to input-group

puiterwijk committed:
Check SSH keys before writing them out

This is needed because Gitolite will abort all ACL and keyfile regeneration
if there is a single invalid key in its keydir.

Signed-off-by: Patrick Uiterwijk <puiterwijk@redhat.com>

pypingou committed:
Fix showing the PR number or New PR button on branches containing a dot

Fixes https://pagure.io/pagure/issue/1361

puiterwijk committed:
Add a validator to check ssh keys on edit

Signed-off-by: Patrick Uiterwijk <puiterwijk@redhat.com>

pypingou committed:
Fix deleting branches containing a dot in their name

puiterwijk committed:
Add test case for SSH key checker

Signed-off-by: Patrick Uiterwijk <puiterwijk@redhat.com>

pypingou committed:
Do not call the API if issue or pull-requests are disabled and we know it

Fixes https://pagure.io/pagure/issue/1364

pypingou committed:
Check if there is something in form.milestone before retrieving it

Fixes https://pagure.io/pagure/issue/1336

pypingou committed:
Use the raw_data check instead of checking if the returned namespace is 'None'

pypingou committed:
Add unit-tests about forking a non-existing repo with a namespace

pypingou committed:
Fix removing all the milestones of a project

puiterwijk committed:
If there's no clamd, skip

Signed-off-by: Patrick Uiterwijk <puiterwijk@redhat.com>

pypingou committed:
Raise a specific error when no PR can be attached to a test run

And do not notify the admins about this, test runs can happen for any
reason, not necessarily because of a PR, so if pagure can't associate
a run with a PR, so be it.

pypingou committed:
Let only project admins create repo

Fixes https://pagure.io/pagure/issue/1335

pypingou committed:
Fix setting name, we want to check the PR settings to load PRs

pypingou committed:
Ensure we always pass two object to the .done() function

Then check if these two objects contain something before playing with
them.

pypingou committed:
Remove redundant code and ensure #<id> and @<user> can be used in the initial comment

pypingou committed:
Drop left over debugging code

pypingou committed:
Fix unit-tests :)

d3prof3t committed:
used correct params for progress bar calculation

pypingou committed:
Drop the use of form.<field>.raw_data since it now seems to work without it

pypingou committed:
Port pagure to flask_wtf 0.13+

pypingou committed:
Drop using `as FlaskForm` where not needed

puiterwijk committed:
Use /dev/stdin instead of -f - for ssh-keygen

-f - is only supported on newer versions of ssh-keygen.

Signed-off-by: Patrick Uiterwijk <puiterwijk@redhat.com>

pypingou committed:
Fix displaying why a PR cannot be merged

Fixes https://pagure.io/pagure/issue/1366

pypingou committed:
Fix figuring out if the milestone of a ticket was edited or not

Basically, before we always added an empty choice to the list of
milestones. This lead to a situation where there was always something
assigned as milestone even when the project had none.
With this change, we first check if there is a milestone before
adding the empty choice to the list of choices and before retrieving
the milestone from the form.

pypingou committed:
Open a session to the DB for every message received and close it afterward

This ensure the data isn't cached and that updating the settings of
a project will be reflected in this service (thus if one updates the
URL, the service will be updated).

Fixes https://pagure.io/pagure/issue/1248

pypingou committed:
Fix the unit-tests now that we know when a milestone was edited

jeremycline committed:
Each test now works in a tempdir

This is motivated by the tests all failing over SSHFS. Each test now
creates its test repositories in a temporary directory which is cleaned
up after each test.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

pypingou committed:
Add a welcome screen to new users of pagure

Fixes https://pagure.io/pagure/issue/1234

pypingou committed:
Add a project's settings to allow turning on/off fedmsg notifications

Fixes https://pagure.io/pagure/issue/1380

pypingou committed:
Add close_status to issues

The list of status is a property of the project and can be used to
specify why/how a ticket was closed.

pypingou committed:
Add possibility to set/remove the close status in the settings page of a project

pypingou committed:
Adjust the issue view to include the close_status information

Fixes https://pagure.io/pagure/issue/1373

pypingou committed:
Add support for close_status in the API while keeping the backward compatibility

pypingou committed:
Fix showing the status of the issue in the issue list

Fixes https://pagure.io/pagure/issue/1381

pypingou committed:
Adjust unit-tests for the changes related to close_status

pypingou committed:
Allow the returned close_status to be None and not 'None'

pypingou committed:
Allow all the forms containing an optional SelectField to return None

Otherwise, when the field was containing None it would be converted
to 'None' this is due to the default coerce being `unicode` in wtforms
cf: http://wtforms.simplecodes.com/docs/0.6/fields.html#wtforms.fields.SelectField

pypingou committed:
Small grammar improvements suggested by @tibbs, thanks!

pypingou committed:
Provide the searched pattern to the issues template

pypingou committed:
Make the search box working and pretty on the issues page

pypingou committed:
Small indentation fix in the issues template

pypingou committed:
Make the welcome screen an alert instead of a panel

pypingou committed:
Drop this flash message, duplicates the welcome screen

pypingou committed:
Make the instance name configurable and reword a little the welcome screen

pypingou committed:
Ensure the user remains logged out if it couldn't be properly set-up

Fixes https://pagure.io/pagure/issue/1388

pypingou committed:
Introduce a default hook/plugin

This hook/plugin is meant to be activated for all projects on pagure
and for the moment, all it does is resetting the merge status of all
open pull-requests upon push to the main branch.

pypingou committed:
Do not include the default hook in the list of plugins

pypingou committed:
Move the logic related to plugins to pagure.lib.plugins and adjust calls

Adjust the unit-tests while at it to cover this change

pypingou committed:
Install the default hook/plugin when creating a new project

Adjust the unit-tests for this change

pypingou committed:
We cannot strip None objects

pypingou committed:
Fix the pagure hook with the close_status change

pypingou committed:
Style fix (tab -> spaces)

pypingou committed:
Do not allow changing the default plugin via the UI

pypingou committed:
Add unit-tests for the default hook/plugin

pypingou committed:
Some more style fixes (tab -> spaces and indentation)

pypingou committed:
Fix issue when running the tests on jenkins with sqlite and log error when querying faitout

pypingou committed:
Add some logging to know how the tests are being run

pypingou committed:
Log a little more info about what returned faitout

pypingou committed:
Fix the faitout url

pypingou committed:
Add the close_status to the new_issue method in pagure.lib

pypingou committed:
Adjust the unit-tests for the changes related to the close_status

pypingou committed:
More unit-test fixes

Ensure the order remains consistent
Use boolean where postgresql expects them

pypingou committed:
Fix bug when changing the status of a ticket via the API

pypingou committed:
Fix search_issues to be backward compatible and search the close_status where desired

pypingou committed:
We're searching closed issues everytime we're not searching open ones

pypingou committed:
Adjust unit-tests to be more flexible with the ordering

pypingou committed:
Fix search issues by status

pypingou committed:
Fix the ordering in the tests

pypingou committed:
Sort the emails when generating the JSON view of an user

pypingou committed:
Fix searching for closed issues

pypingou committed:
Set the session in pagure.ui.app as well to run the tests

pypingou committed:
Add some debugging code

pypingou committed:
Set the status in the form so that it validates

pypingou committed:
Fix docstrings in the default hook

pypingou committed:
No need to rebuild the search query when searching issues

Instead of building a new query, re-use the existing one and adjust it
for the new search criteria.

pypingou committed:
Remove debugging code

pypingou committed:
Add an alembic migration to set the default hook to all existing projects

pypingou committed:
Fix cloning repo which have a branch named 'origin'

Without limiting the replace, the entire string was replaced and we
ended up with a name being '' instead of 'origin'.

pypingou committed:
Adjust the DB model for storing the custom fields of tickets/projects

pypingou committed:
Add methods to pagure.lib to set the custom fields and values on issues

pypingou committed:
Add endpoint to save the custom fields set for a project

pypingou committed:
Adjust the settings page to allow setting custom fields for issues

pypingou committed:
Display the custom fields on tickets and allow updating/saving them

pypingou committed:
Save the changes as we go

pypingou committed:
Fix setting the custom key value

pypingou committed:
Allow filtering issues using the custom fields set-up

pypingou committed:
Adjust the database model with the feedback from @puiterwijk

pypingou committed:
Fix setting custom fields for the adjusted DB model

pypingou committed:
Ensure when the project is deleted, its custom fields are as well

pypingou committed:
Make the UI reflects the key_type (text/boolean)

pypingou committed:
Fix the form so that the values returned are strings as expected

pypingou committed:
Ensure we write the files in UTF-8

pypingou committed:
Only select the close_status when there is one

pypingou committed:
Release 2.7

pypingou committed:
Give an idea about which project is being adjusted

pypingou committed:
Fix deleting custom fields

The way of deleting a custom field is to set it to an empty string.
With the check on length we were always triggering the exception.
Instead of that we should just edit/add what was given and remove
the rest.

pypingou committed:
Fix displaying the input fields properly so that they can be edited

pypingou committed:
Release 2.7.1

pypingou committed:
Do not show the custom key block if the project has not set any

jeremycline committed:
Note that SEND_EMAIL=False does not disable EMAIL_ERROR

I discovered today that when I set ``SEND_EMAIL = False`` in the Pagure
configuration, unhandled exceptions were still emailed to pingou which
is embarassing for me and noise for him. This adds a note to the
configuration documentation so that those who come after me do not make
the same mistake.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

pypingou committed:
Release 2.7.2

pypingou committed:
Delete the old status only at the end

pypingou committed:
Add a _settings field in the user table

This will be used to store global user's settings

pypingou committed:
Add the possibility for the user to edit their settings in their settings page

This allows turning on/off their settings as they wish

pypingou committed:
Notify user of their own action if they asked for it

Fixes https://pagure.io/pagure/issue/1400

pypingou committed:
Fix creating new PR from the page listing the pull-requests

In this page we do two queries:
- Give me all the branch of interest in the fork of the user
- Give me all the branch of interest in the main repo (if the user is
  an admin)

That second request was broken because the `repo_admin` variable was no
longer used (instead if we use the one in flask.g).

pypingou committed:
Fix the RTD and Force push hooks for the change in location of the plugins

Fixes https://pagure.io/pagure/issue/1414

pypingou committed:
Fix left over from copy/paste and code style error pointed out by @vivekanand1101

pypingou committed:
Fix the docstring

pypingou committed:
Make the search case-insensitive when searching issues

Fixes https://pagure.io/pagure/issue/1417

pypingou committed:
Fix the pagination when searching through issues

Fixes https://pagure.io/pagure/issue/1419

pypingou committed:
Fix the "cancel" button when editing a "regular" comment on a pull-request

Fixes https://pagure.io/pagure/issue/1416

pypingou committed:
Let the username be empty if the user isn't logged in

Fixes running the tests when EVENTSOURCE_SOURCE is defined in the
configuration.

Fixes https://pagure.io/pagure/issue/1355

pypingou committed:
Include the close_status in the JSON representation of an issue

Adjust the unit-tests for the corresponding change

pypingou committed:
Load the close_status if there is one set in the JSON repsentation given

pypingou committed:
Let's not print the patch everytime we run - thanks @vivekanand1101

jeremycline committed:
Remove the ``Content-Encoding`` headers from responses

According to RFC7231[0] Content-Encoding is used to indicate the
response has been encoded using, for example, gzip or base64. It was
being used to indicate the character encoding used in responses, rather
than the ``charset=<encoding>`` section of the ``Content-Type`` header.
This is related to, but does not solve,
https://pagure.io/pagure/issue/891.

[0] https://tools.ietf.org/html/rfc7231#section-3.1.2

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

pypingou committed:
Fix creating the release folder for project with a namespace

For project having a namespace the release folder than contains a slash
which means, we need to create two folders instead of one.

Fixes https://pagure.io/pagure/issue/1425

pypingou committed:
When sending email, make the user who made the action be in the From field

Fixes https://pagure.io/pagure/issue/1427

pypingou committed:
When searching groups, search both their name and display name

Fixes https://pagure.io/pagure/issue/1433

jeremycline committed:
Add some encoding/decoding utilities for git files

Users have regularly noticed that some files are being returned with an
incorrect character encoding. This is because the files were not pure
ascii, but did not bear any difinitive markers of the UTF family of
encodings. It tends to fall back to ISO-8859-2 (latin-2) or
Windows-1250. This adds some helpful utilities that will use chardet to
guess the encoding, but will heavily skew the results in favor of utf-8
if the encoding is in doubt. It also modifies the raw file view and the
highlighted code view to use the new utilities.

Fixes https://pagure.io/pagure/issue/891

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

ryanlerch committed:
made search stop clearing status and tags filters

fixes #1432
